compiler/rustc_borrowck/src/diagnostics/move_errors.rs

```diff
@@ -414,11 +414,23 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
             && use_spans.map_or(true, |v| !v.for_closure())
             && !has_complex_bindings
         {
-            err.span_suggestion_verbose(
+            let span = match use_spans {
+                Some(UseSpans::FnSelfUse { var_span, fn_call_span, .. }) => {
+                    // `fn_call_span` is the span of `unwrap()`
+                    // This gets the 'variable' span excluding the unwrap
+                    let span = var_span.until(fn_call_span);
+
+                    // Unfortunately the new span has a "." on the end which makes
+                    // the suggestion format strangely, so we adjust the hi bound manually
+                    span.with_hi(span.hi() - BytePos(1))
+                }
+                _ => span,
```

Comment:
I suspect that this case may never happen (as an Option or Result must always be unwraped, or expected, etc) but am not confident enough to make this a panic or some such. Can change at reviewer discretion

Reply:
I'm not sure, let's keep it as is.
(And set applicability to `MaybeIncorrect` for this case.)